coda/coda_mdstore/presentation.py

```diff
@@ -79,6 +81,33 @@ def getFileHandle(codaId, codaPath):
     )
 
 
+def file_chunk_generator(url):
+    """
+    Download a file and stream it
+    """
+    r = requests.get(url, stream=True)
+    if r.status_code != 200:
+        return
+    for chunk in r.iter_content(1024):
+        yield chunk
+
+
+def zip_file_streamer(urls, identifier):
+    """
+    Stream downloadable zipped file using zipstream
+    """
+    with zipstream.ZipFile(mode='w') as zip_obj:
+        for url in urls:
+            meta_id = identifier.split('/')[-1]
+            filename = '%s/%s' % (meta_id, url.split(meta_id, 1)[-1])
```

Comment:
I _kind_ of wonder if this would be better suited for the `re` module, but I'm not sure it's warranted. But that's just a thought, take it or leave it :smile: